src/styles.py

```diff
@@ -605,6 +614,19 @@ def add_color(self, _button):
 
     @Gtk.Template.Callback()
     def on_delete(self, _button):
+        style_manager = self._style_manager
+        style_model = style_manager.props.style_model
+        file = self.style.get_file()
+        for index, style in enumerate(style_model):
+            if style is not None:
+                file2 = style.get_file()
+                if file2 is not None and file.equal(file2):
+                    stylename = style.get_name()
+                    style_model.remove(index)
+                    style_manager._stylenames.remove(stylename)
+                    style_manager.props.use_custom_style = False
+                    style_manager.props.custom_style = \
+                        style_manager._system_style_name
```

Comment:
This logic is making a few calls, that should rather be in the stylemanager itself imo.